feat(mobile): new settings ui

dvbthien opened the pull request:
## Description

Redesigned settings pages with a more modern and eye-catching interface. 

I have attached images in the Screenshots

<details><summary><h2>Screenshots (if appropriate)</h2></summary>

<!-- Images go below this line. -->

<img src="https://github.com/user-attachments/assets/e5a09760-13af-4fe5-8258-77ac6f141538" width="400" alt="Simulator Screenshot - iPhone SE (3rd generation)">
<img src="https://github.com/user-attachments/assets/a1858fb7-23a6-4da7-911c-56138312f0cd" width="400" alt="Simulator Screenshot - iPhone SE (3rd generation)">
<img src="https://github.com/user-attachments/assets/71aab7b1-80cf-4e24-8a3e-2daade41d417" width="400" alt="Simulator Screenshot - iPhone SE (3rd generation)">
<img src="https://github.com/user-attachments/assets/508537ec-074a-4419-a065-3ebf11098fd7" width="400" alt="Simulator Screenshot - iPhone SE (3rd generation)">
<img src="https://github.com/user-attachments/assets/99638c9e-81bc-442d-8079-e45fed2d5143" width="400" alt="Simulator Screenshot - iPhone SE (3rd generation)">
<img src="https://github.com/user-attachments/assets/4145faa3-fbf4-4a54-998a-eedd7afe689f" width="400" alt="Simulator Screenshot - iPhone SE (3rd generation)">


</details>




## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code

Commit:
implement new settings ui

Commit:
fix lint

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

Commit:
update background color

alextran1502 commented:
Great work, I like the design a lot. There are three things I would like to get your help with

1. Can you keep the original translation?

2. Make sure that the spacing is consistent

<img width="912" alt="image" src="https://github.com/user-attachments/assets/c66af3c8-1245-4d64-8c0e-1d768584ab32" />

3. This header looks out of place

<img width="589" alt="image" src="https://github.com/user-attachments/assets/6bfcddba-e897-40e2-abe3-8ca35a558f6f" />

Commit:
add missing translate

Commit:
reduce add new translation keep the original translation

Commit:
reduce translate for network

Commit:
enhance the space between cards

Commit:
fix the inconsistent spacing in the advanced

dvbthien commented:
Hi @alextran1502 

Regarding the points you mentioned, I'll address each one:

1. I will try to minimize changes to the original translations, but I have added some new translations for certain UI elements.

2. I fixed that issue in my last commit.

3. I changed the header color so it doesn't look out of place. Please see the image below—what do you think?

<img width="356" alt="Screenshot 2025-06-18 at 01 57 19" src="https://github.com/user-attachments/assets/14c6b575-2ef6-4a92-84ec-1b3a1b46f6c6" />

4. I think using the same header style for all segments would be challenging because each segment requires the appropriate icon and title. If you have any ideas, please share them.

5. I fixed that issue in my last commit.

6. I believe the current design for the backup option is good. If you want to use the same segment style, I can try changing it locally so you can better visualize it. What are your thoughts? If you like it or if you have new ideas, please let me know. I have attached images of the backup option with the segment style.

<img width="310" alt="Screenshot 2025-06-18 at 02 15 12" src="https://github.com/user-attachments/assets/85f36d03-a67e-49da-883d-73d19be9b125" /> <img width="387" alt="Screenshot 2025-06-18 at 02 15 22" src="https://github.com/user-attachments/assets/9e98effb-25a3-437b-b82e-4cac861e97f8" />

-------------
In my last commit, I changed the card's background color have more white and reduced the border width. Do you like it?

<img width="387" alt="Screenshot 2025-06-18 at 02 15 22" src="https://github.com/user-attachments/assets/94be227d-22cf-4327-82e6-a6073fbbf6a2" />


Let me know your thoughts. Thank you!

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

alextran1502 commented:
Hi Thien,

3. We are using all capital letters here, so ideally, the font size should be smaller, try 13 or 14
4. You can make them with a header with a title as "Placeholder", I can go through and come up with an appropriate one 
6. Yeah, I like the new option. I wonder if we can swap out the elevated button for a switch like other toggleable options in the settings

> In my last commit, I changed the card's background color have more white and reduced the border width. Do you like it?

Looks good, when I go through the final tests, I can make small adjustments if needed

Commit:
chore: reduce size sub header network

Commit:
chore: new backup page design

Commit:
fix format

Commit:
reduce change translation

Commit:
add header with placehoder

Commit:
fix: triggerDelay updated while widget build

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

Commit:
improve slider style

Commit:
add text style extension for settings

Commit:
clean up foreground backup

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

dvbthien commented:
Hi @alextran1502,

I have made the changes per your suggestions. I have also added a header titled "Placeholder" for all sections. Please review it when you have time. 

Additionally, I have a new design for the backup page. Please see the image below.

 <img width="387" alt="Screenshot 2025-06-18 at 02 15 22" src="https://github.com/user-attachments/assets/66911407-2123-4937-b06e-b8604c842ef5" />

Commit:
use t extension to translate

Commit:
Merge branch 'mobile/new-settings-ui' of https://github.com/dvbthien/immich into mobile/new-settings-ui

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

Commit:
fade on tap for settings page

Commit:
fix lint

Commit:
apply ellipsis to overflowing section header

Commit:
improve: split widgets to reduce rebuilds when settings change

dvbthien committed:
Merge branch 'main' into mobile/new-settings-ui

Commit:
test: new font Mulish

dvbthien commented:
Hi @alextran1502,

Everything has been completed for you to review in the last two commits. If you find it difficult to review, I can revert those changes and create a separate pull request.

1. In the first commit, I split the settings into smaller widgets so that unrelated widgets are not rebuilt and the code logic is easier to understand.

2. In the second commit, I tried a new font and found it very suitable for the application. I pushed it for you to review—if you like it, please let me know.

Have a nice day,
Thanks

alextran1502 commented:
Hi @dvbthien, please keep this PR containing the new design for the settings stuff. Other refactoring should be in their PRs. 

Regarding the fonts, I would like to make this change when needed. It is an important factor in the whole app's look and feel

Commit:
Revert "improve: split widgets to reduce rebuilds when settings change"

This reverts commit 53e1a88b05cdb8b79fd107351ee2a9c350212023.

Commit:
Revert "test: new font Mulish"

This reverts commit c07d0e345ce8b4a217e045885f99e90e0e0d5ca8.

alextran1502 committed:
merge main

alextran1502 commented on mobile/lib/utils/fade_on_tap.dart:
```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/material.dart';
+
+class FadeOnTap extends StatelessWidget {
```
What is this for? I don't notice the different of the components that use this

dvbthien replied:
You can long-press to notice the difference—using `onTap` in `ListTile` triggers the default splash effect. I used `FadeOnTap` to avoid the splash while still providing tap feedback. The default splash effect can be tricky to handle in some areas due to padding or borders.

alextran1502 replied:
Why do we want to remove the default splash behavior?

dvbthien replied:
I used padding in `SettingsCardLayout` to align all items and set the `ListTile` padding to 0. With the default splash behavior, the ripple ignores that padding and can’t be properly clipped where there’s a border. I also chose a `Container` over a `Card` for the settings card to save on performance, since we don’t need elevation—but the container doesn’t clip the splash as cleanly as a card does.

<img 
  src="https://github.com/user-attachments/assets/8f1819fd-681a-4e8b-a0b0-df43b9699d6e" 
  alt="Simulator Screenshot" 
  width="300" 
/>

jrasm91 commented:
This had a lot of back and forth, but I'm not sure what the status is now. There are a _lot_ of merge conflicts so I'm going to close this as stable. If this is something we want to merge, we'll need to rebase the changes and ping Alex to give it another round of review.
